src/main/kotlin/io/github/paulgriffith/kindling/idb/IdbView.kt

```diff
@@ -20,80 +23,145 @@ class IdbView(path: Path) : ToolPanel() {
         rs.getString(3)
     }
 
-    private var tool: IdbTool by Delegates.vetoable(
-        when {
-            "logging_event" in tables -> IdbTool.Log
-            "SYSTEM_METRICS" in tables -> IdbTool.Metrics
-            else -> IdbTool.Generic
-        },
-    ) { _, _, newValue ->
-        try {
-            val newPanel = newValue.openPanel(connection)
-            removeAll()
-            add(newPanel, "push, grow")
-            true
-        } catch (e: Exception) {
-            JOptionPane.showMessageDialog(
-                this,
-                "Unable to open as a ${newValue.name}; ${e.message}",
-                "Error",
-                JOptionPane.ERROR_MESSAGE,
-                FlatSVGIcon("icons/bx-error.svg"),
-            )
-            LOGGER.error("Unable to swap tool to {}", newValue, e)
-            false
-        }
+    private val tabs = TabStrip().apply {
+        trailingComponent = null
+        isTabsClosable = false
     }
 
     init {
         name = path.name
         toolTipText = path.toString()
 
-        add(tool.openPanel(connection), "push, grow")
+        tabs.addTab(
+            tabName = "Tables",
+            component = GenericView(connection),
+            tabTooltip = null,
+            select = true,
+        )
+
```

Comment:
The lazy tab is cool and all, but I think it's more useful to have an IDB initally show its custom view first. Most users expect a log view when opening a log IDB and I think 99% of the time they don't care about the raw db view.